src/univariate/continuous/loguniform.jl

```diff
@@ -0,0 +1,66 @@
+"""
+    LogUniform(a,b)
+
+A positive random variable `X` is log-uniformly with parameters `a` and `b` if the logarithm of `X` is `Uniform(log(a), log(b))`.
+The *log uniform* distribution is also known as *reciprocal distribution*.
+```julia
+LogUniform(1,10)
+```
+External links
+
+* [Log uniform distribution on Wikipedia](https://en.wikipedia.org/wiki/Reciprocal_distribution)
+"""
+struct LogUniform{T<:Real} <: ContinuousUnivariateDistribution
+    a::T
+    b::T
+    LogUniform{T}(a::T, b::T) where {T <: Real} = new{T}(a, b)
+end
+
+function LogUniform(a::T, b::T; check_args=true) where {T <: Real}
+    check_args && @check_args(LogUniform, 0 < a < b)
+    LogUniform{T}(a, b)
+end
+
+LogUniform(a::Real, b::Real) = LogUniform(promote(a, b)...)
+
+function convert(::Type{LogUniform{T}}, d::LogUniform) where {T}
+    LogUniform(T(d.a), T(d.b))::LogUniform{T}
+end
+Base.minimum(d::LogUniform) = d.a
+Base.maximum(d::LogUniform) = d.b
+
+#### Parameters
+params(d::LogUniform) = (d.a, d.b)
+partype(::LogUniform{T}) where {T<:Real} = T
+
+#### Statistics
+
+function mean(d::LogUniform)
+    a, b = params(d)
+    (b - a) / log(b/a)
+end
+function var(d::LogUniform)
+    a, b = params(d)
+    log_ba = log(b/a)
+    (b^2 - a^2) / (2*log_ba) - ((b-a)/ log_ba)^2
+end
+mode(d::LogUniform)   = d.a
+modes(d::LogUniform)  = partype(d)[]
+
+#### Evaluation
+function pdf(d::LogUniform, x::Real)
+    x1, a, b = promote(x, params(d)...) # ensure e.g. pfd(LogUniform(1,2), 1f0)::Float32
+    res = inv(x1 * log(b / a))
+    return insupport(d, x1) ? res : zero(res)
+end
+function cdf(d::LogUniform, x::Real)
+    x1, a, b = promote(x, params(d)...)
+    x2 = clamp(x, a, b)
+    return log(x2 / a) / log(b / a)
```

Comment:
```suggestion
    x2 = clamp(x1, a, b)
    return log(x2 / a) / log(b / a)
```
Alternatively, since `x1`, `a`, and `b` are already of the same type you should be able to use
```suggestion
    x1 = clamp(x1, a, b)
    return log(x1 / a) / log(b / a)
```

Reply:
@devmotion that one is tricky. Tha `clamp` takes care of promoting `x` but does not promote `a,b,`. Consider `a::Int, b::Int, x::Float32`. With the variant, you suggest the `log(b/a)` term will poison everything into `Float64`.

Reply:
Ah no, I didn't suggest removing the `promote` line. Just fixing and changing the two lines below.